python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -487,32 +487,6 @@ def test_options_mode():
     assert xpd.options.mode.copy_on_write == pd.options.mode.copy_on_write
 
 
-# Codecov and Profiler interfere with each-other,
-# hence we don't want to run code-cov on this test.
-@pytest.mark.no_cover
-def test_profiler():
```

Comment:
There's an issue running this test with xdist. I think the problem is that `Profiler` uses `sys.settrace` to collect cudf.pandas calls which doesn't work with the multiple processes created when running the test with xdist.

I think this is true because in the test, `p.per_line_stats == 0` (which means its empty). So the function calls are not being collected.

Reply:
I'm not sure if the reason above is correct because this [test](https://github.com/rapidsai/cudf/pull/16930/files#diff-84df08f4a7d8cb805040b27477cffa48d61aa4eff69e2b1c19e1afd47a9a7315L25-L67), which also uses `Profiler` passes.

Reply:
cc. @galipremsagar @vyasr @mroeschke 

If you guys might know whats going on.

Reply:
I don't think we are thread-safe in our profiler. Is there any way to retain this test and use `xdist`? How much time savings are we having using `xdist` for this test suite?

Reply:
pytest-xdist is multiprocess, not multithreaded, so thread safety shouldn't be an issue. If I don't get to this by EOD, remind me to look into this in a week after I come back.